src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -207,8 +207,8 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
         }
 
         $parameters = $method->getParameters();
-        if (1 !== \count($parameters)) {
-            throw new RuntimeException(sprintf('Invalid handler service "%s": method "%s::__invoke()" must have exactly one argument corresponding to the message it handles.', $serviceId, $handlerClass->getName()));
+        if (0 === \count($parameters)) {
```

Comment:
maybe use getNumberOfRequiredParameters?

Reply:
I changed it, although not sure if it helps against `(Message $message = null)` since you could have `(Message $message = null, Envelope $envelope)`?